Check DEM provided when extracting layers for which necessary

rzinke opened the pull request:
Added function to check that a valid DEM is specified or downloaded when requesting ariaExtract.py to extract 'all' or any of 'bPerpendicular', 'bParallel', 'incidenceAngle', 'lookAngle', 'azimuthAngle', or 'solidEarthTide', for which a DEM is necessary.

Commit:
Check DEM provided when extracting layers for which necessary

pep8speaks commented:
Hello @rzinke! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:



There are currently no PEP 8 issues detected in this Pull Request. Cheers! :beers: 

##### Comment last updated at 2024-07-23 19:08:28 UTC

sssangha committed:
Patch to catch ionosphere

Additionally, this check was failing originally regardless of what metadata layer was specified as the check for `all` layers

sssangha committed:
Adjust list of layers requiring DEM

sssangha committed:
pep8 suggestions implemented

sssangha committed:
Adjust long line

sssangha commented:
Thanks @rzinke!

I issued a patch to capture a bug with the way the `all` keyword was handled. In the initial commit, you had dummy-proofed it by forcing it to be lower-case. But the way it was implemented kicked out all keywords that weren't `all`. I.e. if I specified just SET then an empty list would be passed.

Also, I made some minor tweaks to meet pep8 suggestions. @alexfore , could you please advise on the implementation on this end? Thanks

Here are some simple tests to test this all out:
```
# get products
wget https://gunw-development-testing.s3.us-west-2.amazonaws.com/unavco_2024/S1-GUNW-A-R-124-tops-20180502_20180408-043026-00157W_00018N-PP-9909-v3_0_1.nc .
wget https://gunw-development-testing.s3.us-west-2.amazonaws.com/unavco_2024/S1-GUNW-A-R-124-tops-20180502_20180408-043052-00157W_00019N-PP-7296-v3_0_1.nc .

# test SET extraction without specifying a valid dem
ariaExtract.py -f "products/*.nc" -l solidEarthTide -w test_set_nodem

# test SET extraction with DEM download specified
ariaExtract.py -f "products/*.nc" -l solidEarthTide -w test_set_dem -d Download
```

alexfore commented on tools/bin/ariaExtract.py:
```diff
@@ -142,6 +142,34 @@ def main():
+    # Check whether all necessary inputs were specified.
+    # some products require a DEM to extract -- if any of those are requested,
+    # ensure that a valid DEM is specified
+    if args.layers:
```
Recommend to use the more pythonic version of this: `args.layers is not None:`

alexfore replied:
@rzinke Just this last comment here, thanks!

sssangha replied:
Done, thanks @alexfore !

alexfore commented on tools/bin/ariaExtract.py:
```diff
@@ -142,6 +142,34 @@ def main():
+                  else layer for layer in layers]
+
+        # list of layers requiring DEM for extraction
+        layers_requiring_dem = ['all',
```
Recommend `LAYERS_REQUIRING_DEM` here for constant (constants in UPPER_CASE)

alexfore commented on tools/bin/ariaExtract.py:
```diff
@@ -142,6 +142,34 @@ def main():
+                                'solidEarthTide']
+
+        # check that DEM is specified depending on layers requested
+        intersecting_layers = [layer for layer in layers
```
IMO let's use sets and intersection here (more expressive):
`if len(set(LAYERS_REQUIRING_DEM).intersection(layers)) == 0:`
or declare LAYERS_REQUIRING_DEM as a set (use curly braces), then it will be 
`if len(LAYERS_REQUIRING_DEM.intersection(layers)) == 0:`

alexfore commented on tools/bin/ariaExtract.py:
```diff
@@ -142,6 +142,34 @@ def main():
+        intersecting_layers = [layer for layer in layers
+                               if layer in layers_requiring_dem]
+        if intersecting_layers != []:
+            assert args.demfile is not None, (
```
Should this be an assertion, or should we logically and the assertion with the conditional on the line above then raise an Exception and issue a `LOGGER.error` message here if passes conditional? IMO this would be more simple to understand but same behavior

alexfore reviewed (changes requested):
I have some quick comments above, should be minor changes.

rzinke reviewed (commented):
Thank you both. I will implement the suggested changes, except that the ionosphere layer doesn't need a DEM (I checked last week when modifying the notebook).

Commit:
Error logging and pep8 for DEM check

rzinke commented:
Updates made [0b94a0d](0b94a0dadbbd2e1d0b12dba4e970ce5ec595c564)
@alexfore could you please review?

sssangha committed:
Update layer check to be more pythonic

alexfore reviewed (approved):
Thanks!
